capstone-app/src/app/modules/photos/components/image-container/image-container.component.spec.ts

```diff
@@ -87,4 +87,18 @@ describe('ImageContainerComponent', () => {
     component.ngOnChanges(changesObj);
     expect(component.filter).toBe('');
   });
+
+  it('filter should be active marker when there exists a current active marker', () => {
+    component.city = 'Boston';
+    component.ngOnChanges({});
```

Comment:
Do these tests still work if you replace the direct calls to ngOnChanges with fixture.detectChanges()? If so, that's preferable.

Reply:
Same for other tests.

Reply:
Nope they don't. I believe it's because the code I need execute specifically occurs in ngOnChanges and fixture.detectChanges() technically does not call ngOnChanges.